### DISCUSSION TITLE
Blocking of script tags added in cookie consent.

### DISCUSSION THREAD

**Discussion by benj-webriq** [orestbida/cookieconsent#705] [Q&A]
Hi, I am new to this cookie consent thing, I have a question. 
When the user accept/decline the cookie does this library automatically disable/enable the script of that cookie or do I have to implement the blocking of script myself?

**Comment by tilcordes** [comment]
Hi, the cookieconsent-tool doesn't automatically block your scripts if the user disagrees with their usage. 
In the official [documentation](https://cookieconsent.orestbida.com/advanced/manage-scripts.html) of the tool it is very well described how to block your scripts.

There are basically two ways:
- "data-category" tag option: This involves adding a variable in the script tag that you want to block.
- event-listeners: In my view, this method has some advantages. Event listeners can detect every possible user action, such as initially allowing cookies and later in the same session changing their preference to not allow them. The “data-category” variable might not detect these changes.

All the listeners and how to use them are well described in the documentation.

I hope this helps!
